language-extensions/dotnet-core-CSharp/sample/regex/README.md

```diff
@@ -179,7 +179,7 @@ To execute .NET C# code, the user SID [S-1-15-2-1](https://docs.microsoft.com/en
 Create an external library for the RegEx code.
 
 ```sql
-CREATE EXTERNAL LIBRARY regex
+CREATE EXTERNAL LIBRARY [regex.dll]
```

Comment:
Few questions to get more context for this change:
1. Why is .dll needed? 
2. Should the value used match the filename referenced in `CONTENT` and if so, should the value be `regexsample.dll`? 3. Is the value used here case sensitive?

Reply:
> Few questions to get more context for this change:
> 
> 1. Why is .dll needed?
> 2. Should the value used match the filename referenced in `CONTENT` and if so, should the value be `regexsample.dll`?
> 3. Is the value used here case sensitive?

1. When we don't add `.dll` suffix for the csharp libraries, it is created as a file with no extension under the external library folder. Not sure where but I think later we are looking for a file with `.dll` as its extension.
2. It doesn't have to match the file name that was provided in the content. In fact, we are creating an external library that is known to the SQL Server from content that we have built before. In this example, we create the external library `regex.dll` from the built library `RegexSample.dll` which was created by compiling `RegexSample.cs`. After that SQL Server understands the library by its provided name: here, `regex.dll`.
3). Even under a database with case-sensitive/accent-sensitive collation, the name of the library is **not** case-sensitive: 
![image](https://github.com/microsoft/sql-server-language-extensions/assets/77026206/76cb7f90-1cd2-49ec-b41b-bf9f195bb8b9)

Reply:
@SicongLiu2000, please comment if you know where we expect the library file to end with .dll (have dll extension).

Reply:
Sounds like the following is the root cause of needing to update these docs:

> Not sure where but I think later we are looking for a file with .dll as its extension.

This must be C# specific behavior because other languages R (.R), Java (.jar), Python (.py) don't require their respective file extensions in the value used in `CREATE EXTERNAL LANGUAGE`